### ISSUE TITLE
'logfmt' parser doesn't parse empty string as expect

### ISSUE THREAD

**Issue by GaoyangWang** [fluent/fluent-bit#1737]
## Bug Report

**Describe the bug**
When I config 'logfmt' parser to parse the input content, such as: key1="" key2 key3="this is value", it will parse key1 as same as key2, a boolean value "true". For the expected 'logfmt' behaviour, key1 should be an empty string instead of a boolen value.

**To Reproduce**
- Rubular link if applicable:
- Example log message if applicable:
```
key1="" key2 key3=123 key4="this is a text"
```
- Steps to reproduce the problem:
0. just use the hello_world example
1. config parser as 'logfmt'
2. add a lua filter to handle the input parser's result
3. the data (messagepack) pass to filter, key1 is stored as a boolean value.

**Expected behavior**
'logfmt' should support to parse an empty string. 

**Screenshots**

**Your Environment**

* Version used: 1.3.2
* Configuration:
* Environment name and version (e.g. Kubernetes? What version?): 
* Server type and version:
* Operating System and version: Ubuntu 18.04.3
* Filters and plugins: Lua filter and 'logfmt' parser plugin

**Additional context**

**Comment by GaoyangWang** [comment]
I think the root cause is in src/flb_parser_logfmt.c line 173:
```
if (value_len == 0) {
    msgpack_pack_true(tmp_pck);
}
```

**Comment by edsiper** [comment]
thanks for reporting and troubleshooting the root cause of the issue.

I've pushed a fix into GIT Master and on 1.3 branches for the next v1.3.4 release this week. The output will look as follows:

```json
{"date":1576020907.000000,"key1":"","key2":null,"key3":"123","key4":"this is a text"}
```
